tests/model/model.cpp

```diff
@@ -112,6 +112,99 @@ TEST(Model, add_components) {
     EXPECT_EQ(e, a);
 }
 
+TEST(Model, count) {
+    libcellml::Model m;
+    libcellml::Component c1, c2;
+    c1.setName("child1");
+    c2.setName("child2");
+
+    EXPECT_EQ(0, m.componentCount());
+
+    m.addComponent(c1);
+    m.addComponent(c2);
+    EXPECT_EQ(2, m.componentCount());
+}
+
+TEST(Model, contains) {
+    libcellml::Model m;
+    libcellml::Component c1, c2;
+
+    EXPECT_FALSE(m.containsComponent("child1"));
+
+    m.addComponent(c1);
+    m.addComponent(c2);
+    EXPECT_TRUE(m.containsComponent("child2"));
+}
+
+TEST(Model, remove) {
+    libcellml::Model m;
+    libcellml::Component c1, c2;
+    c1.setName("child1");
+    c2.setName("child2");
+    m.addComponent(c1);
+    m.addComponent(c2);
+
+    m.removeComponent(0);
+    EXPECT_EQ(1, m.componentCount());
+    EXPECT_THROW(m.removeComponent(1), std::runtime_error);
+
+    m.addComponent(c1);
+    m.addComponent(c1);
+    m.removeComponent("child1");
+    EXPECT_EQ(1, m.componentCount());
```

Comment:
I would expect the above `removeComponent` to only remove one of the `c1` copies from the model and therefore the component count would be 2 in this case.